single-line-diagram/single-line-diagram-core/src/test/java/com/powsybl/sld/iidm/TestCase15GraphWithVoltageIndicator.java

```diff
@@ -53,54 +55,66 @@ public boolean isPowered() {
         }
     }
 
-    private DiagramLabelProvider withFullBusInfoProvider;
+    private ParamBuilder.LabelProviderFactory withFullBusInfoProviderFactory;
 
-    private DiagramLabelProvider withIncompleteBusInfoProvider;
+    private ParamBuilder.LabelProviderFactory withIncompleteBusInfoProviderFactory;
 
     @BeforeEach
     public void setUp() throws IOException {
         network = Networks.createNetworkWithFiveBusesFourLoads();
         graphBuilder = new NetworkGraphBuilder(network);
 
-        withFullBusInfoProvider = new DefaultDiagramLabelProvider(network, componentLibrary, layoutParameters) {
+        withFullBusInfoProviderFactory = new DefaultLabelProviderFactory() {
             @Override
-            public Optional<BusInfo> getBusInfo(BusNode node) {
-                Objects.requireNonNull(node);
-                BusInfo result = null;
-                switch (node.getId()) {
-                    case "bbs21":
-                        result = new BusVoltageInfo(false);
-                        break;
-                    case "bbs1":
-                    case "bbs13":
-                        result = new BusVoltageInfo(true, "Top", null, Side.RIGHT);
-                        break;
-                    case "bbs22":
-                    case "bbs23":
-                        result = new BusVoltageInfo(false, null, "Bottom", Side.LEFT);
-                        break;
-                }
-                return Optional.ofNullable(result);
+            public LabelProvider create(Network network, ComponentLibrary componentLibrary, LayoutParameters layoutParameters, SvgParameters svgParameters) {
+                return new DefaultLabelProvider(network, componentLibrary, layoutParameters, svgParameters) {
```

Comment:
you're creating here a new `DefaultLabelProviderFactory`, overriding the method create. This is an anonymous inner class which extends `DefaultLabelProviderFactory`. But there's nothing left of `DefaultLabelProviderFactory` as you override its only method. The benefit of a functional interface is that you can define it with a lambda, which leads to a more readable code, see below.

Could you simplify this? and also all the other anonymous inner classes you defined in unit tests.

```java
        withFullBusInfoProviderFactory = (network, componentLibrary, layoutParameters, svgParameters) ->
                new DefaultLabelProvider(network, componentLibrary, layoutParameters, svgParameters) {
                    @Override
                    public Optional<BusInfo> getBusInfo(BusNode node) {
                        (...)
                    }
                };
```